### DISCUSSION TITLE
where is the shiori docker database? How to backup and restore

### DISCUSSION THREAD

**Discussion by sunteta** [go-shiori/shiori#446] [Q&A]
How do you back up the shiori under docker, where do the database files exist?
How to backup and restore

thanks! please help

Has been using it for years and can't leave now. A lot of data is important.

**Comment by rmayergfx** [comment]
i am also searching for this. i renamed the original container to shiori_bkp and setup a new one.
There you can add a folder inside your  on your docker folder , e.g. shiori (/volume1/docker/shiori) this will be used to redirect the database 
/volume1/docker/shiori
on the volume settings add your newly created folder you can select it like the one you created inside the docker folder:
/docker/shiori
then set the Mount-Path to
/srv/shiori
Now when you start shiori for the first time the database shiori.db will be created. After the first added bookmark you will find also 2 new folders archive and thumb inside this folder.

Now take a full backup of your container shiori with Docker>Setup>Export container contents and settings !
Marius has a great explanation on this: [https://mariushosting.com/synology-how-to-export-and-import-a-docker-container/](url)

Save this file on a save place and copy it to your local pc in a temporary folder. File looks like shiori.syno.txz
Extract this file with 7-zip or any other archiver  you have already installed and which can handle this file-type.
After the first extract you see inside the folder a file called image.tar, expand this also.
now all your data is located in \image\srv\shiori
archive
thumb
shiori.db

**_Now stop the container shiori !_**
go to your folder /volume1/docker/shiori and make a new folder, eg with name "fresh" and move all files from the new setup inside this folder. Then copy the folders archive, thumb and the shiori.db to your folder /volume1/docker/shiori.
Start the container again and  you will find all your already existing data.
